### PR TITLE
Fix: FK reading grade of 0 < x < 1 collapses to 0 (normalize to 1)

### LINKED ISSUE #1497: Bug: FK Reading Grade of 0 < x < 1 Collapses to 0 (Should Normalize to 1 for Compliance)
### Please give us a description of what happened.

When calculating reading grade level via `DaveChild\TextStatistics\TextStatistics::fleschKincaidGradeLevel()`, any value above 0 but below 1 is floored to 0 in multiple places:
- [admin/class-ajax.php](https://github.com/equalizedigital/accessibility-checker/blob/16e43b9e2d4bf37043a1edbcbcbfec2256925a45/admin/class-ajax.php#L55-L106)
- [includes/classes/class-summary-generator.php](https://github.com/equalizedigital/accessibility-checker/blob/16e43b9e2d4bf37043a1edbcbcbfec2256925a45/includes/classes/class-summary-generator.php#L258-L329)
- Includes rest-api file (`get_readability_data` method) that will be added when sidebar work is merged.

This collapses legitimate simple-grade content (e.g., FK=0.5, FK=0.8) to grade 0, which misrepresents compliance and flags passing content incorrectly as failing.

### Please describe what you expected to happen and why.

FK grade levels above 0 but below 1 (FK=0.5, FK=0.8) should be normalized to 1—not allowed to collapse to 0. Values >= 1 should continue to use `floor()`. This ensures grade levels aren’t misrepresented and simple text is correctly passed for compliance.

**Suggested normalization:**
| Input FK float | Expected Output |
|---------------|-----------------|
| 0.0           | 0               |
| 0.01–0.99     | 1               |
| 1.0           | 1               |
| 1.2           | 1               |
| 9.9           | 9               |
| 10.0          | 10              |

This should apply everywhere grade level is normalized (AJAX, summary generator, REST API). Adding automated unit tests to enforce this boundary is recommended.

### How can we reproduce this behavior?

1. Create content which will have a FK grade between 0.01 and 0.99 (e.g., a simple sentence with short words).
2. Run Accessibility Checker to scan the content and view calculated reading grade.
3. Observe that it says not enough content to determine the grade.

### PR DESCRIPTION
**Pull request by SteveJonesDev**
Fixes #1497

## Description

`floor()` applied directly to a Flesch-Kincaid grade in the range (0, 1) silently collapses it to `0`. That `0` is then interpreted as "not enough content to determine a reading level," so content with a legitimately simple FK grade (e.g. 0.5 or 0.8) is misrepresented as uncalculable and incorrectly flagged as failing.

## What changed

Introduces `edac_normalize_fk_grade( float $fk_grade ): int` in `includes/helper-functions.php`:

- Returns `0` when the library returns ≤ 0 (truly not enough content)
- Returns `max(1, floor($fk_grade))` otherwise — so any positive sub-1 value becomes grade 1

Replaces the raw `floor()` pattern at all four affected call sites:

| File | Location |
|------|----------|
| `admin/class-ajax.php` | Classic metabox readability (line ~83) |
| `admin/class-ajax.php` | Sidebar AJAX readability (line ~674) |
| `includes/classes/class-summary-generator.php` | `calculate_content_grade()` |
| `includes/classes/class-rest-api.php` | `get_readability_data()` |

## Tests

Adds `tests/phpunit/helper-functions/NormalizeFkGradeTest.php` with 11 data-driven cases covering: exact zero, negatives, the 0.01–0.99 boundary, 1.0, mid-grade values, and grades above 9.

## Checklist

- [x] PR is linked to the main issue in the repo
- [x] Tests are added that cover changes

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved simplified summary and readability reading-level calculations by normalizing Flesch–Kincaid grade values more consistently before applying “too high”/failure logic, yielding more accurate results for fractional and very low grade values.
* **Tests**
  * Added automated PHPUnit coverage for the grade-normalization behavior, including zero/negative inputs and boundary/rounding cases.

### CODE REVIEW

File: includes/classes/class-rest-api.php [hunk 1]
```diff
@@ -1077,7 +1077,7 @@ private function get_readability_data( $post_id ) {
 		$simplified_summary_grade = 0;
 		if ( class_exists( 'DaveChild\TextStatistics\TextStatistics' ) ) {
 			$text_statistics          = new \DaveChild\TextStatistics\TextStatistics();
-			$simplified_summary_grade = (int) floor( $text_statistics->fleschKincaidGradeLevel( $simplified_summary ) );
+			$simplified_summary_grade = edac_normalize_fk_grade( $text_statistics->fleschKincaidGradeLevel( $simplified_summary ) );
 		}
 
 		$simplified_summary_grade_failed      = $simplified_summary_grade >= 9;
```
[the remark this answers is not in the document]
Comment (pattonwebz) [reply includes/classes/class-rest-api.php:1083]: Intentionally not addressing this in PR #1725 — we've opened a dedicated issue [#1797](https://github.com/equalizedigital/accessibility-checker/issues/1797) to fix all the inconsistent grade-9 thresholds across the codebase (including this one and the `class-ajax.php:669` case) with proper references to WCAG 3.1.5.

**Review by pattonwebz — APPROVED** [review]
I tested this and it is working for the most part. I did run into the issue that gemini called out about possible false value instead of a float. I tackled the AI review comments about it and added the test cases to cover `false` and `null` scenarios.

I also changed the since tags to be x.x.x and am opening a PR that will add a tool to fill their version numbers in as part of the release prep phase.
